### PR TITLE
feat(kube-proxy): use a proxy for api-rest, etcd and loki operations for the kubectl-plugin

### PR DESCRIPTION
**Pull request by tiagolobocastro**
feat: add kube-proxy library crate

Rename k8s-proxy crate into kube-forward crate.
Add kube-proxy, a helper library to setup clients using the lower-level kube-forward library.
The current client support is via the tower Configuration for an openapi client.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

feat(kubectl-plugin): make use of the kube-proxy with api-rest

When the rest url is not specifically specified, use the kube-proxy library to create
the openapi client which talks to our api-rest service.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

feat(kubectl-plugin): use kube-proxy for etcd dump

Uses the kube-proxy library to create a port-forward to the etcd pod.
Also add an etcd dump command since it's useful for debugging.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

chore: silence tungstenite traces

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

feat(kubectl-plugin): use kube-proxy for the loki queries

Similar to api-rest and etcd, use the kube-proxy to collect logs from loki.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

**Comment by vharsh** [comment]
Post merging this PR, [this](https://github.com/openebs/mayastor-extensions/pull/11) can be merged.

**Review by Abhinandan-Purkait — APPROVED** [review]

File: k8s/forward/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+/// A kubernetes namespace.
+/// If None, the default is "default".
+#[derive(Clone)]
+pub(crate) struct NameSpace(Option<String>);
+impl NameSpace {
+    /// Returns the configured namespace or the default.
+    pub(crate) fn name_any(&self) -> String {
+        let default = "default".to_string();
```
Comment (vharsh) [inline k8s/forward/src/lib.rs:112]: Todo later: The default namespace should ideally be picked up from the kubeconfig or the flags(`--namespace`)..

**Review by vharsh — APPROVED** [review]
lgtm

File: k8s/forward/src/lib.rs (hunk 1 above)
Comment (tiagolobocastro) [reply k8s/forward/src/lib.rs:112]: This is just the default for the type, the kubectl crate passes the namespace down as a parameter, so it should work like that already

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**